contracts/cw721-base/helpers.ts

```diff
@@ -0,0 +1,433 @@
+  // decreaseAllowance: (recipient: string, amount: string) => Promise<string>
+  // transferFrom: (owner: string, recipient: string, amount: string) => Promise<string>
+}
+
+interface CW721Contract {
+  // upload a code blob and returns a codeId
+  upload: () => Promise<number>
+
+  // instantiates a cw721 contract
+  // codeId must come from a previous deploy
+  // label is the public name of the contract in listing
+  // if you set admin, you can run migrations on this contract (likely client.senderAddress)
+  instantiate: (codeId: number, initMsg: Record<string, unknown>, label: string, admin?: string) => Promise<CW721Instance>
+
+  use: (contractAddress: string) => CW721Instance
+}
+
+
+const CW721 = (client: SigningCosmWasmClient): CW721Contract => {
+  const use = (contractAddress: string): CW721Instance => {
+/*    
+    const balance = async (account?: string): Promise<string> => {
+      const address = account || client.senderAddress;  
+      const result = await client.queryContractSmart(contractAddress, {balance: { address }});
+      return result.balance;
+    };
+*/
+    const allowance = async (owner: string, spender: string): Promise<AllowanceResponse> => {
+      return client.queryContractSmart(contractAddress, {allowance: { owner, spender }});
+    };
+
+    const allAllowances = async (owner: string, startAfter?: string, limit?: number): Promise<AllAllowancesResponse> => {
+      return client.queryContractSmart(contractAddress, {all_allowances: { owner, start_after: startAfter, limit }});
+    };
+
+    const allAccounts = async (startAfter?: string, limit?: number): Promise<readonly string[]> => {
+      const accounts: AllAccountsResponse = await client.queryContractSmart(contractAddress, {all_accounts: { start_after: startAfter, limit }});
+      return accounts.accounts;
+    };
+
+    const minter = async (): Promise<any> => {
+      return client.queryContractSmart(contractAddress, {minter: { }});
+    };
+
+    const contractInfo = async (): Promise<any> => {
+      return client.queryContractSmart(contractAddress, {contract_info: { }});
+    };
+
+    const nftInfo = async (token_id: string): Promise<any> => {
+      return client.queryContractSmart(contractAddress, {nft_info: { token_id }});
+    }  
+
+    const allNftInfo = async (token_id: string): Promise<any> => {
+      return client.queryContractSmart(contractAddress, {all_nft_info: { token_id }});
+    } 
+
+    // TODO: Need help here
+    const ownerOf = async (token_id: string): Promise<any> => {
+        return await client.queryContractSmart(contractAddress, {owner_of: {token_id}});
+    }
+
+
+/*
+    const tokenInfo = async (): Promise<any> => {
+      return client.queryContractSmart(contractAddress, {token_info: { }});
+    };
+*/
+    // mints tokens, returns ?
+    const mint = async (token_id: string, owner: string, name:string, description:string): Promise<string> => {
```

Comment:
this is defined as `description?: string` in the interface above. good to be consistent. (and nice to add image url here as well)